src/Avalonia.Controls/TreeViewItem.cs

```diff
@@ -166,30 +168,95 @@ protected override void OnKeyDown(KeyEventArgs e)
         {
             if (!e.Handled)
             {
-                switch (e.Key)
+                Func<TreeViewItem, bool>? handler =
+                    e.Key switch
+                    {
+                        Key.Left => ApplyRecursivelyIfCtrl(FocusAwareCollapseItem, e.KeyModifiers),
+                        Key.Right => ApplyRecursivelyIfCtrl(ExpandItem, e.KeyModifiers),
```

Comment:
I don't see where only Left/Right is handled to expand/collapse the focused node. This was the old functionality. The new logic only applies if the Ctrl key is held and I don't see anywhere else where Left/Right keys are used. Am I missing something?

Reply:
Nevermind, I see how this works now... I don't know why I can't mark as resolved though since I created this comment. Feel free to close this and others to cleanup the comments.

Reply:
Should I rename the function to make that clearer that it does the actual and does it recursively if ctrl is held?

Reply:
The renamed function definitely makes it more clear, thanks! Local function naming really is up to the dev. It doesn't affect anything else so whatever you think best.